.github/workflows/scripts/e2e/gmc_test.sh

```diff
@@ -0,0 +1,199 @@
+       exit 1
+   fi
+
+   echo "Checking response results, make sure the output is reasonable. "
+   local status=false
+   if [[ -f $LOG_PATH/curl_chatqna.log ]] && \
+   [[ $(grep -c "billion" $LOG_PATH/curl_chatqna.log) != 0 ]]; then
+       status=true
+   fi
+   if [ $status == false ]; then
+       echo "Response check failed, please check the logs in artifacts!"
+       exit 1
+   else
+       echo "Response check succeed!"
+   fi
+}
+
+function init_chatqna() {
+    # Copy manifest from GenAIExamples
+    mkdir -p $(pwd)/config/manifests
+    download_manifests "$(pwd)/config/manifests"
+    cp $(pwd)/config/gmcrouter/gmc-router.yaml -p $(pwd)/config/manifests
+    export NAMESPACE="chatqa"
+    # replace the mount dir "path: /mnt/model" with "path: $CHART_MOUNT"
+    find . -name '*.yaml' -type f -exec sed -i "s#path: /mnt#path: $MOUNT_DIR#g" {} \;
+    # replace the repository "image: opea/*" with "image: $IMAGE_REPO/opea/"
+    find . -name '*.yaml' -type f -exec sed -i "s#image: opea/*#image: $IMAGE_REPO/opea/#g" {} \;
+    # set huggingface token
+    # find . -name '*.yaml' -type f -exec sed -i "s#insert-your-huggingface-token-here#$(cat /home/$USER_ID/.cache/huggingface/token)#g" {} \;
+    find . -name '*.yaml' -type f -exec sed -i "s#insert-your-huggingface-token-here#$(cat /home/$USER_ID/.cache/huggingface/token)#g" {} \;
+    # replace namespace "default" with real namespace
+    find . -name '*.yaml' -type f -exec sed -i "s#default.svc#$NAMESPACE.svc#g" {} \;
+}
+
+
+
+function wait_until_pod_ready() {
+    echo "Waiting for the $1 to be ready..."
+    max_retries=3000
+    retry_count=0
+    while ! is_pod_ready $2 $3; do
+        if [ $retry_count -ge $max_retries ]; then
+            echo "$1 is not ready after waiting for a significant amount of time"
+            exit 1
+        fi
+        echo "$1 is not ready yet. Retrying in 10 seconds..."
+        sleep 10
+        output=$(kubectl get pods -n $2)
+        # Check if the command was successful
+        if [ $? -eq 0 ]; then
+          echo "Successfully retrieved $1 information:"
+          echo "$output"
+        else
+          echo "Failed to retrieve $1 information"
+          exit 1
+        fi
+        retry_count=$((retry_count + 1))
+    done
+}
+
+function is_pod_ready() {
+    if [ "$2" == "gmc-controller" ]; then
+      pod_status=$(kubectl get pods -n $1 -o jsonpath='{.items[].status.conditions[?(@.type=="Ready")].status}')
+    else
+      pod_status=$(kubectl get pods -n $1 -l app=$2 -o jsonpath='{.items[].status.conditions[?(@.type=="Ready")].status}')
+    fi
+    if [ "$pod_status" == "True" ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+
+function validate_codegen() {
+    echo "todo"
+}
+
+function download_manifests() {
+  #api_url="https://api.github.com/repos/opea-project/GenAIExamples/contents/ChatQnA/kubernetes/manifests"
+  api_url="https://api.github.com/repos/opea-project/GenAIInfra/contents/manifests/ChatQnA"
```

Comment:
I wonder why we need download here. In a github workflow, the source code will be checkout to ${{github.workspace}}. This variable can be used in github workflow. So you can "cd ${{github.workspace}}/manifests/ChatQnA" and then use the manifest files there.

Reply:
do we plan to move these manifests to https://github.com/opea-project/GenAIExamples  ?  this is used to handle for this future change. Plz let me know if this is not the case then I will switch to use local repo files